src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { USER_ID } from '../../api/todos';
+
+import { ErrorMessages } from '../../types/ErrorTypes';
+import { Todo } from '../../types/Todo';
+
+import { handleError } from '../../utils/handleError';
+
+interface Props {
+  onTempTodo: (todo: Todo | null) => void;
+  onErrorMessage: (type: ErrorMessages) => void;
+  onAdd: (todo: Todo) => Promise<void>;
+  isAllCompleted: boolean;
+  todos: Todo[];
+}
+
+export const TodoHeader: React.FC<Props> = ({
+  isAllCompleted,
+  todos,
+  onAdd,
+  onErrorMessage,
+  onTempTodo,
+}) => {
+  const [title, setTitle] = useState('');
+  const [isSubmiting, setIsSubmiting] = useState(false);
+
+  const todoInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    todoInput.current?.focus();
+  }, [title, isSubmiting, todos]);
```

Comment:
Move the `useEffect `to the end of code
Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.